dss/api/search.py

```diff
@@ -0,0 +1,41 @@
+import json
+import os
+
+from elasticsearch_dsl import Search
+from flask import request
+
+from .. import DSS_ELASTICSEARCH_INDEX_NAME, DSS_ELASTICSEARCH_DOC_TYPE
+from .. import get_logger
+from ..util import connect_elasticsearch
+
+
+def list():
+    es_client = connect_elasticsearch(os.getenv("DSS_ES_ENDPOINT"), get_logger())  # TODO Use a connection manager
+    get_logger().debug("Searching for: %s", request.values["query"])
+    query = json.loads(request.values["query"])
+    response = Search(using=es_client,
+                      index=DSS_ELASTICSEARCH_INDEX_NAME,
+                      doc_type=DSS_ELASTICSEARCH_DOC_TYPE).query("match", **query).execute()
+    return {"query": query, "results": format_results(request, response)}
+
+
+def post(query: dict):
+    get_logger().debug("Received posted query: %s", json.dumps(query, indent=4))
+    es_client = connect_elasticsearch(os.getenv("DSS_ES_ENDPOINT"), get_logger())  # TODO Use a connection manager
+    response = Search(using=es_client,
+                      index=DSS_ELASTICSEARCH_INDEX_NAME,
+                      doc_type=DSS_ELASTICSEARCH_DOC_TYPE).update_from_dict(query).execute()
```

Comment:
Here update_from_dict is being used to set a fully defined / self-contained Elasticsearch query. In the case of .query("match", **query) a portion of a query is already being specified by "match".